sklearn/linear_model/tests/test_perceptron.py

```diff
@@ -40,8 +38,9 @@ def predict(self, X):
         return np.sign(self.project(X))
 
 
-def test_perceptron_accuracy():
-    for data in (X, X_csr):
+@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
+def test_perceptron_accuracy(csr_container):
+    for data in (X, csr_container(X)):
         clf = Perceptron(max_iter=100, tol=None, shuffle=False)
         clf.fit(data, y)
         score = clf.score(data, y)
```

Comment:
I think we can simply use parametrize to avoid the loop inside this test as suggested below.

```suggestion
@pytest.mark.parametrize(container", [np.array] + CSR_CONTAINERS)
def test_perceptron_accuracy(container):
        data = container(X)
        clf = Perceptron(max_iter=100, tol=None, shuffle=False)
        clf.fit(data, y)
        score = clf.score(data, y)
```

Reply:
Haha, we had the same suggestion concurrently :)